java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1858,71 +1543,44 @@ public ExpressionTree newAnnotatedParameterizedIdentifier(
-          children.add(rest.getLastChild());
-        }
+      for (Tuple<ExpressionTree, Optional<InternalSyntaxToken>> rest : rests.get()) {
+        //FIXME SONARJAVA-547 commas should be handled.
+        initializers.add(rest.first());
       }
     }
-    children.add(closeBraceToken);
-
-    return new NewArrayTreeImpl(ImmutableList.<ExpressionTree>of(), initializers.build(), children).completeWithCurlyBraces(openBraceToken, closeBraceToken);
+    return new NewArrayTreeImpl(ImmutableList.<ExpressionTree>of(), initializers.build()).completeWithCurlyBraces(openBraceToken, closeBraceToken);
   }
 
   public QualifiedIdentifierListTreeImpl newQualifiedIdentifierList(TypeTree qualifiedIdentifier, Optional<List<Tuple<InternalSyntaxToken, TypeTree>>> rests) {
     ImmutableList.Builder<TypeTree> qualifiedIdentifiers = ImmutableList.builder();
-    List<AstNode> children = Lists.newArrayList();
 
     qualifiedIdentifiers.add(qualifiedIdentifier);
-    children.add((AstNode) qualifiedIdentifier);
 
     if (rests.isPresent()) {
       for (Tuple<InternalSyntaxToken, TypeTree> rest : rests.get()) {
         qualifiedIdentifiers.add(rest.second());
-        children.add(rest.first());
-        children.add((AstNode) rest.second());
       }
     }
 
-    return new QualifiedIdentifierListTreeImpl(qualifiedIdentifiers.build(), children);
+    return new QualifiedIdentifierListTreeImpl(qualifiedIdentifiers.build());
   }
 
   public ArrayAccessExpressionTreeImpl newArrayAccessExpression(Optional<List<AnnotationTreeImpl>> annotations, InternalSyntaxToken openBracketToken, ExpressionTree index,
                                                                 InternalSyntaxToken closeBracketToken) {
-    ArrayAccessExpressionTreeImpl result = new ArrayAccessExpressionTreeImpl(openBracketToken, index, closeBracketToken);
-    if (annotations.isPresent()) {
-      result.prependChildren(annotations.get());
-    }
-
-    return result;
+    // FIXME SONARJAVA-547 Handle annotations
+    return new ArrayAccessExpressionTreeImpl(openBracketToken, index, closeBracketToken);
   }
 
   public NewClassTreeImpl newClassCreatorRest(ArgumentListTreeImpl arguments, Optional<ClassTreeImpl> classBody) {
-    List<AstNode> children = Lists.newArrayList();
-    children.add(arguments);
-
-    if (classBody.isPresent()) {
-      children.add(classBody.get());
-    }
-
-    return new NewClassTreeImpl(arguments.openParenToken(), arguments, arguments.closeParenToken(), classBody.isPresent() ? classBody.get() : null,
-      children.toArray(new AstNode[0]));
+    return new NewClassTreeImpl(arguments.openParenToken(), arguments, arguments.closeParenToken(), classBody.isPresent() ? classBody.get() : null);
   }
 
   public ExpressionTree newIdentifierOrMethodInvocation(Optional<TypeArgumentListTreeImpl> typeArguments, JavaTree identifierToken, Optional<ArgumentListTreeImpl> arguments) {
+    // FIXME SONARJAVA-547 Handle type arguments
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)